src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java

```diff
@@ -234,6 +235,15 @@ protected void submit(StaplerRequest req, StaplerResponse rsp) throws IOExceptio
         }
     }
 
+    @Override
+    protected @Nonnull String getSuccessfulDestination() {
+        FolderComputation<I> computation = getComputation();
+        if (computation != null) {
```

Comment:
It does seem true that `computation` will not be null. However you still need to wait for the computation you _scheduled_ to start. Otherwise you may be rendering an empty page from the initial (no-parent) dummy `FolderComputation` which has no log file. It depends on the exact timing of the first JS call to `progressText` after the browser renders the redirect.